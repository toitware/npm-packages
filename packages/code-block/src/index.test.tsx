import { render } from "@testing-library/react";
import React from "react";
import { CodeBlock } from ".";

jest.mock("use-codemirror");

test("properly sanitizes code", () => {
  const result = render(
    <CodeBlock
      code={`

      main:

      map ::= {
        1234: "Siri",
        2345: "John",
        3456: "Sue"
        }

      map.do: | id name |
        print "$name has ID $id"

  
    `}
    />
  );
  expect(result.container).toMatchSnapshot();
});
