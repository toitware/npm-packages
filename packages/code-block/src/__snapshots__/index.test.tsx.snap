// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`properly sanitizes code 1`] = `
<div>
  <div
    class="makeStyles-code-1"
  >
    <pre
      class="3024-day makeStyles-loadingCode-2"
    >
      main:

map ::= {
  1234: "Siri",
  2345: "John",
  3456: "Sue"
  }

map.do: | id name |
  print "$name has ID $id"
    </pre>
  </div>
</div>
`;
